python/lib/dependabot/python/update_checker.rb

```diff
@@ -261,22 +269,23 @@ def latest_version_finder
         )
       end
 
+      def poetry_based?
+        pyproject && !poetry_details.nil?
+      end
+
       def poetry_library?
```

Comment:
I've been adding a bunch of specs here and I think I ended up finding a bug. We only detect libraries in the case of poetry, so if a Python library is using standard python and has a requirement in `pyproject.toml` like this:

```toml
dependencies = [
  "requests~=1.0.0"
]
```

Then with this PR dependabot will propose an update like this:

```toml
dependencies = [
  "requests~=2.19.1"
]
```

instead of widening the requirement, like we do for Poetry

```toml
dependencies = [
  "requests>=1,<3"
]
```

I'll push all the new specs, including a failing one for this, and will try to wrap this up tomorrow.

Reply:
Fixed by https://github.com/dependabot/dependabot-core/pull/5661/commits/ea1ad25ef843b0d641ac931e78c2dafab754a8bc!